services/registry/src/validation/validators/upstreamValidator.ts

```diff
@@ -0,0 +1,24 @@
+import { take, map, filter } from "rxjs/operators";
+
+import gqlObjects$ from "../../sync/sync-service";
+import { AgogosObjectConfig, UpstreamConfig } from "../../sync/object-types";
+
+const validateUpstream = async (
+    source: string,
+    name: string,
+    spec: AgogosObjectConfig
+): Promise<void> => {
+    const upstreams = await gqlObjects$
+        .pipe(
+            map(x => x.upstreams || {}),
```

Comment:
Does this `map`->`take(1)` pattern used in all validators not block deployment when agogos is up but doesn't have any CRDs deployed?

Reply:
There is `startWith` [here](https://github.com/Soluto/agogos/blob/7c60e86149d9e84920a6431a2b313f90dfaf9492/services/registry/src/sync/sync-service.ts#L70).
So it shouldn't be empty ever